pallets/transaction-multi-payment/src/lib.rs

```diff
@@ -218,7 +218,8 @@ pub mod pallet {
 		/// When currency is set, fixed fee is withdrawn from the account to pay for the currency change
 		///
 		/// Emits `CurrencySet` event when successful.
-		#[pallet::weight((<T as Config>::WeightInfo::set_currency(), DispatchClass::Normal, Pays::No))]
+		// REVIEW: You would want to charge before the transaction is executed, no?
```

Comment:
This would be preferable way. But here we need to pay the fee in the currency which is being set with this call.  

That was the reason to manually withdraw a fee within the call. 

Any tip if there is easy to pay the fee before the is executed but using the selected currency ?

Reply:
We should have a filter on this call and it should get rejected by a node if you don't have enough to pay for the fee after the fact so spam should not be possible (only on rpc side which is reasonable I think)

Reply:
You could probably introduce a SignedExtension that inspects the call and checks the balance for the configured currency, yeah.